### PR TITLE
Added settings options for allowed network types

### PR DESCRIPTION
**Pull request by poisdeux**
This allows the user to select other network connection types,
besides WiFi, for downloading media files.

The "Any" network type is added to allow downloading over USB network connections and probably bluetooth as well (didn't test that, as I do not have bluetooth on a device running Kodi).

### CODE REVIEW

File: app/src/main/java/org/xbmc/kore/ui/AlbumDetailsFragment.java [hunk 1]
```diff
@@ -156,13 +157,7 @@ public static AlbumDetailsFragment newInstance(AlbumListFragment.ViewHolder vh)
...
 
     @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-    }
-
-    @Override
-    @TargetApi(21)
-    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    protected View createView(LayoutInflater inflater, ViewGroup container) {
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/AlbumDetailsFragment.java]: lint error here. Keep the TargetApi line.

**Comment by SyncedSynapse** [comment]
@poisdeux 
In the future please don't auto indent the whole file, only the lines you touch, so as to reduce the number of changes. Looking through all these changes is difficult and easy to miss something.

Also, make sure you always run lint (`./gradlew build` should do it) so as to catch the maximum number of issues.

File: app/src/main/java/org/xbmc/kore/ui/AlbumDetailsFragment.java (hunk 1 above)
Comment (poisdeux) [inline app/src/main/java/org/xbmc/kore/ui/AlbumDetailsFragment.java]: Fixed.

**Comment by poisdeux** [comment]
Sorry about the diffs introduced by auto indenting. As my programming consists of a lot of copy/pasting I got used to auto indenting over the years. I do it without even thinking about it. I'll try to be more careful.

**Comment by SyncedSynapse** [comment]
I'm probably going to create a "Advanced settings" section and move this setting to there, later.

**Pull request merged**